### PR TITLE
New trino version 428

### PR DESCRIPTION
**Pull request by maltesander**
# Description

- added support for version 428
- removed support for versions 377, 387, 395, 396, 403
- build opa-authorizer from https://github.com/bloomberg/trino/tree/add-open-policy-agent and adapted to our docker flow (pack jar, load to nexus ...)
- added trino version upload script

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
- [ ] Changes are OpenShift compatible
- [ ] All added packages (via microdnf or otherwise) have a comment on why they are added
- [ ] Things not downloaded from Red Hat repositories should be mirrored in the Stackable repository and downloaded from there
- [ ] All packages should have (if available) signatures/hashes verified
- [ ] Does your change affect an SBOM? Make sure to update all SBOMs
```

### CODE REVIEW

File: trino/upload_new_trino_version.sh [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+# It is probably redundant to check both the checksum and the signature but it's cheap and why not
+echo "Validating SHA1 Checksum"
+if ! (sha1sum "${bin_file}" | cut -d " " -f 1 | diff -Z - "${bin_file}.sha1"); then
+  echo "ERROR: The SHA1 sum does not match"
+  exit 1
+fi
+
+echo "Validating signature"
+echo '--> NOTE: Make sure you have added the public key (https://app.nuclino.com/Stackable/Knowledge-Base/Finding-GPG-keys-for-Maven-packages-6b7b0324-8f0f-4b0f-a6e8-aa97cea5512c) to GPG: https://www.apache.org/info/verification.html'
```
Comment (nightkr) [inline trino/upload_new_trino_version.sh:47]: _Which_ PGP key? GPG also doesn't seem to like grabbing it...

```
n@emeritus:~/Downloads$ gpg --receive-keys C328250FE23A2420814521EC0EB69F76FD171538
gpg: key 0EB69F76FD171538: no user ID
gpg: Total number processed: 1
n@emeritus:~/Downloads$ echo $?
2
```

Comment (nightkr) [reply trino/upload_new_trino_version.sh:47]: Without a solid web of trust or some form of key pinning then GPG signatures are worth no more than a plain old SHA2 signature.

File: trino/upload_new_trino_version.sh [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+trap cleanup EXIT
+
+cd "$WORK_DIR" || exit
+
+bin_file=trino-server-$VERSION.tar.gz
+
+echo "Downloading Trino (this can take a while, it is intentionally downloading from a slow mirror that contains all old versions)"
+curl --fail -LOs "https://repo1.maven.org/maven2/io/trino/trino-server/${VERSION}/${bin_file}"
+curl --fail -LOs "https://repo1.maven.org/maven2/io/trino/trino-server/${VERSION}/${bin_file}.asc"
+curl --fail -LOs "https://repo1.maven.org/maven2/io/trino/trino-server/${VERSION}/${bin_file}.sha1"
```
[the remark this answers is not in the document]
Comment (maltesander) [reply trino/upload_new_trino_version.sh:37]: Only sha1 and md5

File: trino/upload_new_trino_version.sh (hunk 1 above)
Comment (maltesander) [reply trino/upload_new_trino_version.sh:47]: Yeah im for removing this, @Jimvin likes to have it in. I dont see the point currently.

File: trino/upload_new_trino_version.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (nightkr) [reply trino/upload_new_trino_version.sh:37]: Is it worth raising a ticket about this upstream?

[the remark this answers is not in the document]
Comment (maltesander) [reply trino/upload_new_trino_version.sh:37]: We have a couple of tickets we could raise upstream now :D but yeah fine with me!

File: trino/upload_new_trino_version.sh (hunk 1 above)
Comment (nightkr) [reply trino/upload_new_trino_version.sh:47]: (`--no-default-keyring` is required to force it to import the key even if it already exists in your default keyring.. gpgv ignores the default keyring so no issue there)

Comment (nightkr) [reply trino/upload_new_trino_version.sh:47]: This should hopefully also remove the manual intervention and Nuclino dependency..

Comment (maltesander) [reply trino/upload_new_trino_version.sh]: https://github.com/stackabletech/docker-images/pull/487/commits/21230a6a25575baff6d634ca64fbce1ff949532e

**Review by nightkr — APPROVED** [review]
LGTM

**Pull request merged**